File: Classes/VB.Net/UMSConstraint.vb
Imports YamlDotNet.Serialization

Namespace UMS

    ''' <summary>
    ''' A class to store Fact-Based Modelling Constraint details in Constrolled Natural Language.
    '''   NB Is for constraints not otherwise covered by the Primary Key and Uniqueness Constraints within UMS.Type etc.
    ''' </summary>    
    Public Class FBMConstraint
        Inherits UMS.ModelElement

        ''' <summary>
        ''' Verbalisation of the Fact-Based Modelling Constraint.
        ''' E.g. The following is a Join Subset Constraint:
        '''
        ''' If some Booking has some Seat then
        ''' that Booking is for some Session that is at some Cinema that contains some Row that contains that Seat;
        ''' 
        ''' </summary>
        ''' <returns></returns>
        Public Property Verbalisation As String

    End Class

End Namespace

File: Classes/Delphi/UMSConstraint.pas
unit UMSConstraint;

interface

uses UMSModelElement;

type
  /// <summary>A fact-based modelling constraint expressed in controlled natural language.</summary>
  TUMSFBMConstraint = class(TUMSModelElement)
  private
    FVerbalisation: string;
  public
    property Verbalisation: string read FVerbalisation write FVerbalisation;
  end;

implementation

end.
